static/app/views/performance/browser/webVitals/utils/queries/rawWebVitalsQueries/useProjectRawWebVitalsTimeseriesQuery.tsx

```diff
@@ -0,0 +1,142 @@
+
+type Props = {
+  enabled?: boolean;
+  tag?: Tag;
+  transaction?: string | null;
+};
+
+export type WebVitalsScoreBreakdown = {
+  cls: SeriesDataUnit[];
+  fcp: SeriesDataUnit[];
+  fid: SeriesDataUnit[];
+  lcp: SeriesDataUnit[];
+  total: SeriesDataUnit[];
+  ttfb: SeriesDataUnit[];
+};
+
+export const useProjectRawWebVitalsTimeseriesQuery = ({
+  transaction,
+  tag,
+  enabled = true,
+}: Props) => {
+  const pageFilters = usePageFilters();
+  const location = useLocation();
+  const organization = useOrganization();
+  const projectTimeSeriesEventView = EventView.fromNewQueryWithPageFilters(
+    {
+      yAxis: [
+        'p75(measurements.lcp)',
+        'p75(measurements.fcp)',
+        'p75(measurements.cls)',
+        'p75(measurements.ttfb)',
+        'p75(measurements.fid)',
+        'count()',
+      ],
+      name: 'Web Vitals',
+      query:
+        'transaction.op:pageload' +
+        (transaction ? ` transaction:"${transaction}"` : '') +
+        (tag ? ` ${tag.key}:"${tag.name}"` : ''),
+      version: 2,
+      fields: [],
+      interval: getInterval(pageFilters.selection.datetime, 'low'),
+      dataset: DiscoverDatasets.METRICS,
+    },
+    pageFilters.selection
+  );
+
+  const result = useGenericDiscoverQuery<
+    {
+      data: any[];
+      meta: MetaType;
+    },
+    DiscoverQueryProps
+  >({
+    route: 'events-stats',
+    eventView: projectTimeSeriesEventView,
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...projectTimeSeriesEventView.getEventsAPIPayload(location),
+      yAxis: projectTimeSeriesEventView.yAxis,
+      topEvents: projectTimeSeriesEventView.topEvents,
+      excludeOther: 0,
+      partial: 1,
+      orderby: undefined,
+      interval: projectTimeSeriesEventView.interval,
+    }),
+    options: {
+      enabled: pageFilters.isReady && enabled,
+      refetchOnWindowFocus: false,
+    },
+    referrer: 'api.performance.browser.web-vitals.timeseries',
+  });
+
+  const data: WebVitalsScoreBreakdown = {
+    lcp: [],
+    fcp: [],
+    cls: [],
+    ttfb: [],
+    fid: [],
+    total: [],
+  };
+
+  result?.data?.['p75(measurements.lcp)']?.data.forEach((interval, index) => {
+    const lcp: number = result?.data?.['p75(measurements.lcp)']?.data[index][1][0].count;
+    const fcp: number = result?.data?.['p75(measurements.fcp)']?.data[index][1][0].count;
+    const cls: number = result?.data?.['p75(measurements.cls)']?.data[index][1][0].count;
+    const ttfb: number =
+      result?.data?.['p75(measurements.ttfb)']?.data[index][1][0].count;
+    const fid: number = result?.data?.['p75(measurements.fid)']?.data[index][1][0].count;
+    // This is kinda jank, but since events-stats zero fills, we need to assume that 0 values mean no data.
+    // 0 value for a webvital is low frequency, but not impossible. We may need to figure out a better way to handle this in the future.
```

Comment:
That's an awful lot of repetition! Iterating though `['lcp', 'fcp', 'cls', 'ttfb', 'fid']` would be way nicer

Reply:
Also applies to all the code below

Reply:
I've cleaned this up to be much shorter now!